### ISSUE TITLE
ValueChangeListener not working on deleting chips

### ISSUE THREAD

**Issue by ngonzalezpazFC** [FlowingCode/ChipFieldAddon#32]
As far as I tested, the problem occurs only while deleting elements. The valueChangeListener does not catch the event in these cases, neither having a DataProvider nor using setAvailableItems method:

- Removing a chip that was added programatically via setValue method.
- Removing a new created chip, not provided by the DataProvider.

**Comment by javier-godoy** [comment]
@ngonzalezpazFC Can you test again with the latest changes in master?

**Comment by ngonzalezpazFC** [comment]
The lastest changes solved this issue.

**Comment by ngonzalezpazFC** [comment]
Close #32

**Comment by javier-godoy** [comment]
Note: this was fixed by #35 and/or #36

**Comment by ngonzalezpazFC** [comment]
Manually deleting an item added with setValue(), as the following code shows, was not detected before by the value change event , but now it does:
```
chipfield.addValueChangeListener(e -> {
		Notification n = new Notification("Value changed");
		n.open();
	});
chipfield.setValue(Arrays.asList(new Planet("Test")));
```
Also, deleting a a chip added manually (chip not provided by the DataProvider assigned to the chipfield), is now triggering the value change event.
